### ISSUE TITLE
Field level @auth rules are not applied

### ISSUE THREAD

**Issue by mansdahlstrom1** [aws-amplify/amplify-cli#7431]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.15.4

### Amplify CLI Version

4.51.4

### What operating system are you using?

Mac

### Amplify Categories

api

### Amplify Commands

push, Not applicable

### Describe the bug

TLDR; Top level model @auth blocks field-level @auth rules.

Given the following GQL schema I'm expecting that owners are allowed to  update the `handled` and `isRevealed` field in the modal with a mutation. Instead AppSync returns the following error: 

To debug this further I've tried to modify the generated VTL template and concluded that my request never reaches the VTl template. Instead it seems like the request is blocked before the VTL Templates is reached.

See my Schema bellow:

```gql
type MyModel
  @model(subscriptions: null)
  @auth(
    rules: [
      { allow: owner, ownerField: "userId", identityClaim: "sub", operations: [read] } # Cognito User Pools (owner)
      { allow: private, operations: [read] } # Cognito User Pools
      { allow: private, provider: iam }
    ]
  ) {
  id: ID!
    @auth(
      rules: [{ allow: owner, ownerField: "userId", identityClaim: "sub", operations: [update] }]
    )
  eventId: ID!
  teamId: ID!
  userId: ID!
  #  ...other read only props

  handled: Boolean
    @auth(
      rules: [{ allow: owner, ownerField: "userId", identityClaim: "sub", operations: [update] }]
    )

  isRevealed: Boolean
    @auth(
      rules: [{ allow: owner, ownerField: "userId", identityClaim: "sub", operations: [update] }]
    )
}

```

I've been using the AppSync console to test all of this with the following queries

```gql

mutation ShouldFail {
  updateMyModel(input: {id: "f551baff-cd63-4e31-aa3b-73c87bec8783", isRevealed: true, winnings: 100}) {
    id
    isRevealed
    winnings
  }
}

mutation ShouldSucceed {
  updateMyModel(input: {id: "f551baff-cd63-4e31-aa3b-73c87bec8783", isRevealed: true, handled:false}) {
    id
    isRevealed
    winnings
  }
}
```

### Expected behavior

Test query `shouldFail`  above should fail with an unauthorised message. Test query `shouldSucceed` above should work. 

### Reproduction steps

1. Copy schema and deploy with Cognito UserPools setup
2. Run the test queries described above  

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line
type MyModel
  @model(subscriptions: null)
  @auth(
    rules: [
      { allow: owner, ownerField: "userId", identityClaim: "sub", operations: [read] } # Cognito User Pools (owner)
      { allow: private, operations: [read] } # Cognito User Pools
      { allow: private, provider: iam }
    ]
  ) {
  id: ID!
    @auth(
      rules: [{ allow: owner, ownerField: "userId", identityClaim: "sub", operations: [update] }]
    )
  eventId: ID!
  teamId: ID!
  userId: ID!
  #  ...other read only props

  handled: Boolean
    @auth(
      rules: [{ allow: owner, ownerField: "userId", identityClaim: "sub", operations: [update] }]
    )

  isRevealed: Boolean
    @auth(
      rules: [{ allow: owner, ownerField: "userId", identityClaim: "sub", operations: [update] }]
    )
}


```

</details>

### Log output

<details>

```
# Put your logs below this line
{
  "data": {
    "updateMyModel": null
  },
  "errors": [
    {
      "path": [
        "updateMyModel"
      ],
      "data": null,
      "errorType": "Unauthorized",
      "errorInfo": null,
      "locations": [
        {
          "line": 30,
          "column": 3,
          "sourceName": null
        }
      ],
      "message": "Not Authorized to access updateMyModel on type Mutation"
    }
  ]
}

```

</details>

**Comment by SwaySway** [comment]
@mansdahlstrom1 
Based on your example it looks like what you are describing is deny-by-default behavior.
The field does add authorization on the field itself though it does not deny access to any other field (ex. winnings).
 We are tracking these improvements as a part of the next version of the GraphQL transformer.

Closing this issue in favor of the RFCs listed below.

`@auth` Improvements RFC: https://github.com/aws-amplify/amplify-cli/issues/1043
 vNext Transformer RFC: https://github.com/aws-amplify/amplify-cli/issues/6217

**Comment by mansdahlstrom1** [comment]
Hi @SwaySway and thanks for the response. Both my queries return `unauthorised` currently, not the otherway around.  If a field level `@auth` exists on a model, then it should allow for update on the top model right. I'm not sure this is the same as deny-by-default, but i might be wrong. 

This error seems to more that if there is no `@auth update` on the model, field level `@auth` is ignored. From reading the RFC's you linked i did not see that this use case is covered. Should this be added somehow to the RFC?

**Comment by mansdahlstrom1** [comment]
Hi again! 

Just took another look at this and not sure if i should put more info the RFCs. I'm stil not sure that deny-by-default is the right "folder" to put this in.

Deny by default as i understand it is the following:
```
// Open by default
type Generic @model {
  id: ID!
}


// Not affected by deny-by-default behaviour
type Generic @model @auth( rules: [ { allow: owner } ] ) {
  id: ID!
}
```

The issue here is what i put in the TLDR in the description
> TLDR; Top level model @auth blocks field-level @auth rules.

So in my case it seems like the the top model blocks any requests at an IAM level, before the VTL Kicks in.

So to reiterate: Should this be added to the RFC somehow? Does it fit in with the RFC? Or is this supposed to be supported?
